File: Problems/0532.K-Diff-Pairs-In-An-Array/KDiffPairsInAnArray_test.py

from KDiffPairsInAnArray import Solution
import pytest
from copy import deepcopy

SOL = Solution()

TEST_SUITS =  [
    ([6, 3, 5, 7, 2, 3, 3, 8, 2, 4], 2, 5)
]

@pytest.mark.parametrize(
    "nums, k, ans", 
    deepcopy(TEST_SUITS)
)
def test(nums, k, ans):
    assert SOL.findPairs(nums, k) == ans
    assert SOL.findPairsV2(nums, k) == ans
    

File: Problems/0532.K-Diff-Pairs-In-An-Array/KDiffPairsInAnArray_test.go

package leetcode

import (
	"testing"
)

func Test(t *testing.T) {
	testSuits := []struct {
		name string
		nums []int
		k    int
		ans  int
	}{
		{"Example 1", []int{6, 3, 5, 7, 2, 3, 3, 8, 2, 4}, 2, 5},
	}

	for _, test := range testSuits {
		got := findPairs(test.nums, test.k)

		if got != test.ans {
			t.Errorf("[%v] expect '%v', got '%v' ", test.name, test.ans, got)
		}
	}
}
